orttraining/orttraining/core/graph/gradient_schema_defs.cc

```diff
@@ -1749,6 +1749,10 @@ Return true if all elements are true and false otherwise.
         // which are only used for maintain topological order
         for (size_t i = 0; i < ctx.getNumOutputs(); ++i) {
           propagateElemTypeFromInputToOutput(ctx, i + 1, i);
+          auto typeProto = ctx.getInputType(i + 1);
```

Comment:
Without this, unknown shape will be propagated as scalar shape `{}`.

Reply:
Seems like something that propagateShapeFromInputToOutput should do?

---
In reply to: [425912809](https://github.com/microsoft/onnxruntime/pull/3823#discussion_r425912809) [](ancestors = 425912809)

Reply:
I agree its a bug in ONNX. I will create a task for fixing that in ONNX repo (I feel it's a long term solution). Sounds ok to you?

---
In reply to: [425965978](https://github.com/microsoft/onnxruntime/pull/3823#discussion_r425965978) [](ancestors = 425965978,425912809)

Reply:
My fix is in [https://github.com/onnx/onnx/pull/2778](https://github.com/onnx/onnx/pull/2778). It's tested against my local ONNXRuntime code.

---
In reply to: [425999675](https://github.com/microsoft/onnxruntime/pull/3823#discussion_r425999675) [](ancestors = 425999675,425965978,425912809)